pyproject.toml

```diff
@@ -2,5 +2,25 @@
 build-backend = "setuptools.build_meta"
 requires = ["setuptools", "wheel"]
 
+[tool.coverage.run]
+branch = true
+source = ["sceptre"]
+
+[tool.coverage.report]
+exclude_lines = [
+  # Have to re-enable the standard pragma
+  "pragma: no cover", # Don't complain about missing debug-only code:
+  "def __repr__",
+  "if self.debug", # Don't complain if tests don't hit defensive assertion code:
+  "raise AssertionError",
+  "raise NotImplementedError", # Don't complain if non-runnable code isn't run:
+  "if 0:",
+  "if __name__ == .__main__.:", # Don't complain about mypy-specific code
+  "if TYPE_CHECKING:",
+]
+ignore_errors = true
+fail_under = 90.0
+show_missing = true
+
 [tool.pytest.ini_options]
-addopts = "tests/ --ignore=env/ --ignore=venv/ --junitxml=test-results/junit.xml -s"
+addopts = "-s --cov --cov-report term --cov-report term-missing --cov-report xml"
```

Comment:
sorry, i don't really know how this toml file works.  does running tox automatically generated coverage reports?

Reply:
This toml file is just a way of configuring many tools in one place. This particular part is the equivalent of having a separate `pytest.ini` file. Many tools in the Python ecosystem have been adopting the ability to add configuration to this file. You can see this repo for a list of such tools: https://github.com/carlosperate/awesome-pyproject

As to the specific configuration listed here, this makes it so that any time `pytest` is run, it automatically generates a coverage report. And since `tox` runs `pytest` yes, that does generate a coverage report. 😄

Reply:
Any reason you didn't move configs in `tox.ini` into this  toml file?

Reply:
I could, I've just been hesitant to move that particular config yet in my projects. According to [the documentation](https://tox.readthedocs.io/en/latest/example/basic.html#pyproject-toml-tox-legacy-ini), it's a little bit of a hack at the moment, and intentionally so. Basically you have to write all of the config into a single big multi-line string, still using the INI format. They do have plans for native TOML config in the future, but it's just not there yet.

So it would look something like this:
```toml
[tool.tox]
legacy_tox_ini = """
[tox]
minversion = 3.23.0
isolated_build = True
envlist = py{36,37,38,39},flake8
skip_missing_interpreters = True

[testenv]
deps =
     -rrequirements/prod.txt
     -rrequirements/dev.txt
commands = pytest {posargs}

[testenv:flake8]
commands = flake8 .
"""
```
Maybe not the worst thing in the world. But my personal preference so far has been to wait until they adopt native TOML configuration. Especially when the tox files [can tend to grow a bit larger](https://github.com/godaddy/tartufo/blob/main/tox.ini).

Reply:
ok, let's just leave as-is then.